src/plugins/data/server/search/search_service.ts

```diff
@@ -243,26 +243,41 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
           };
 
           const dfsService: DataFramesService = {
-            get: () => this.dfCache.get(),
+            get: (name: string) => {
+              const df = this.dfsCache.get(name);
+              if (df) {
+                const sessionId = this.sessionCache.get(df?.name);
+                if (sessionId) {
+                  df.meta = { ...df.meta, sessionId };
+                }
+              }
+              return df;
+            },
             set: async (dataFrame: IDataFrame) => {
-              if (this.dfCache.get() && this.dfCache.get()?.name !== dataFrame.name) {
-                scopedIndexPatterns.clearCache(this.dfCache.get()!.name, false);
```

Comment:
Don't need to `clearCache()` here right? Think we cleared it in the `public` folder